lib/python/rose/suite_clean.py

```diff
@@ -102,11 +102,37 @@ def _clean(self, suite_name, only_items=None):
                     if item:
                         loc_rel = os.path.join(suite_dir_rel, item)
                     locs.append(os.path.join(item_root, loc_rel))
+                    roots.add(item_root)
             if self.host_selector.is_local_host(auth):
+                # Clean relevant items
                 for loc in locs:
                     loc = os.path.abspath(env_var_process(loc))
                     for name in sorted(glob(loc)):
                         engine.fs_util.delete(name)
+                # Clean empty directories
+                # Change directory to root level to avoid cleaning them as well
+                # For cylc suites, e.g. it can clean up to an empty "cylc-run/"
+                # directory.
+                for root in sorted(roots):
+                    cwd = os.getcwd()
+                    if root:
+                        try:
+                            os.chdir(env_var_process(root))
+                        except OSError:
+                            continue
+                    # Reverse sort to ensure that e.g. "share/cycle/" is
+                    # cleaned before "share/"
+                    for name in sorted(self.CLEANABLE_PATHS, reverse=True):
```

Comment:
I'm struggling to come up with a specific example with my (apparently) currently limited brain space, rather a worry based on previous experience with this sort of thing going wrong (you may remember the test battery problem where the test worked for you but not for me because our usernames appeared at different points in a sorted list of results :grimacing:). As test battery's working and I can't come up with a specific example other than this has a potential to be a pain in some unforeseen combination of entries that may never exist I'll merge this in.